### DISCUSSION TITLE
Tested the app for ~20+ Users, some users keep getting disconnecting and reconnecting.

### DISCUSSION THREAD

**Discussion by prayaslashkari** [edumeet/edumeet#776] [Q&A]
Connection is not at all stable for some users. Any fixes regarding this?

**Comment by Alex-YddeR** [comment]
> Connection is not at all stable for some users. Any fixes regarding this?

What is the LastN Configuration and what is the configuration of server and its load. are these users are from mobile or desktop

**Comment by prayaslashkari** [comment]
LastN = 100
MobileLastN = 100
Server Config
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
Memory - 16gb
Browser Used - Chrome

Users were from both mobile and desktop.

**Comment by so010** [comment]
LastN = 100
MobileLastN = 100

These are quite high values - at least to me - you really want to display 100 users at the same time on mobile? I doubt that this performs on mobile devices. Even for classic devices I would begin with LastN 11,15 or 17 and 1 or 2 for MobileLastN.
Client bottlenecks are not detected automatically yet so you need to somehow make a decision what is minimum performance you need for participating.

**Comment by vpalmisano** [comment]
I wouldn't go over 20-25 participants on desktop and 4 on mobile.

**Comment by prayaslashkari** [comment]
So like lastN is basically the participants visible but like their is no bar to how many participants can be their in the meeting right? So suppose i bar the lastN to 25, but 30 users can join the meeting right?

Also if we are to scale the architecture how do we do that? I mean jitsi and zoom, these platforms do have a lot of people that can be their in the meeting? And are visible right?

@so010 @vpalmisano

**Comment by so010** [comment]
Don't know how many participants can be visible at the same time in Jitsi but Zoom has default 25 depending on client/OS and GPU accelleration up to 50 participants visible. We just increased performance and it looks promising. I'm quite positive that soon we will operate at the same level as Zoom.
And yes there can be more participants in the meeting ;)

**Comment by prayaslashkari** [comment]
@so010 Yours answers were quite helpful and insightful, Thank you! I've couple of more questions. SORRY! I am just trying to understand things here, so that our team will be able take desired steps.

Suppose in a meeting we have 100 users,

1) Max 25 LastN, means only 25 users will be visible at a time, but their can be 100 people in the meeting?
2) How are we suppose to see next set 25 users? By making a custom spotlight function?
3) I've seen many webinar being held on zoom, there are like more than 100 users, host able to go to different pages, to see next set of people.
4) What we did here, was to set LastN to 100, create a custom pagination function (paginate via spotlight peer array) which lets you see 14 people at once, as soon as we move to next page 14 people more and so on. But I guess this isnt the best approach and optimized.
5) So, we are not showing 100 users in one page. We have modified the grid accordingly.

**Comment by so010** [comment]
> * Max 25 LastN, means only 25 users will be visible at a time, but their can be 100 people in the meeting?

Yes, I had meetings with up to 150 participants until now.

> * How are we suppose to see next set 25 users? By making a custom spotlight function?
> * I've seen many webinar being held on zoom, there are like more than 100 users, host able to go to different pages, to see 
next set of people.

You can spotlight from the 3 dots in participant list whoever you want. Slide to "next page" of participants is not supported (yet) - In my opinion this is not a good function in Zoom so I doupt we will implement this in that way.


> * What we did here, was to set LastN to 100, create a custom pagination function (paginate via spotlight peer array) which lets you see 14 people at once, as soon as we move to next page 14 people more and so on. But I guess this isnt the best approach and optimized.
> * So, we are not showing 100 users in one page. We have modified the grid accordingly.

 Very thankful for any solutions please invite us to see!

**Comment by so010** [comment]
move to discussions... https://github.com/edumeet/edumeet/discussions
